### PR TITLE
test: add testing for generic builder multi subject

### PR DESCRIPTION
**Pull request by asraa**
Signed-off-by: Asra Ali <asraa@google.com>

Linking issue on multi subject verification https://github.com/slsa-framework/slsa-github-generator/issues/609

Part of https://github.com/slsa-framework/slsa-verifier/issues/133

### CODE REVIEW

File: main_test.go [hunk 1]
```diff
@@ -362,7 +384,7 @@ func Test_runVerify(t *testing.T) {
 						t.Fatal(err)
 					}
 					for _, f := range builderDir {
-						if f.IsDir() {
+						if f.IsDir() && semver.Compare(minversion, f.Name()) <= 0 {
```
Comment (naveensrinivasan) [inline main_test.go:387]: A comment here would be helpful as to the reason for `semver.Compare`

File: main_test.go [hunk 2]
```diff
@@ -343,17 +363,19 @@ func Test_runVerify(t *testing.T) {
...
-				}
-				for _, f := range testdataDir {
-					if f.IsDir() {
-						// These are the builder subfolders
-						builders = append(builders, f.Name())
+				builders := tt.builders
+				if len(builders) == 0 {
+					testdataDir, err := ioutil.ReadDir("./testdata")
+					if err != nil {
+						t.Fatal(err)
```
Comment (naveensrinivasan) [inline main_test.go:372]: A suggestion would be to have `t.Errorf` instead of `t.Fatal`

File: main_test.go [hunk 3]
```diff
@@ -343,17 +363,19 @@ func Test_runVerify(t *testing.T) {
...
-				if err != nil {
-					t.Fatal(err)
-				}
-				for _, f := range testdataDir {
-					if f.IsDir() {
-						// These are the builder subfolders
-						builders = append(builders, f.Name())
+				builders := tt.builders
+				if len(builders) == 0 {
+					testdataDir, err := ioutil.ReadDir("./testdata")
```
Comment (naveensrinivasan) [inline main_test.go:370]: A suggestion would be to have `./testdata` as a constant.

The same  is being used here in next few lines below `ioutil.ReadDir(fmt.Sprintf("./testdata/%s", builder))`

File: main_test.go (hunk 2 above)
Comment (asraa) [reply main_test.go:372]: I think that's fair, I didn't realize that `ReadDir` would return an empty list and I could still continue execution. Thanks!

**Review by laurentsimon — APPROVED** [review]
Awesome, thanks!

**Pull request merged**